ecommerce/api_test.py

```diff
@@ -675,6 +677,25 @@ def test_get_full_price_coupon_product_set():
     }
 
 
+def test_bulk_assign_product_coupons():
+    """
+    bulk_assign_product_coupons should pair emails with available coupons, assign the coupons to those
+    emails, and group them by a bulk assignment record.
+    """
+    emails = sorted(["abc@example.com", "def@example.com", "ghi@example.com"])
+    product_coupons = CouponEligibilityFactory.create_batch(len(emails))
+    paired_email_coupon_assignments = list(zip(emails, product_coupons))
+
+    bulk_assignment, result_iter = bulk_assign_product_coupons(emails, product_coupons)
```

Comment:
Can you change the inputs to be generators so we can test that the iterable is only iterated over once?